nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2485 @@
+"""
+Download statistical maps available on Neurovault (http://neurovault.org).
+"""
+# Author: Jerome Dockes
+# License: simplified BSD
+
+import os
+import warnings
+import traceback
+from copy import copy, deepcopy
+import shutil
+import re
+import json
+from glob import glob
+from tempfile import mkdtemp
+from collections import Container
+try:
+    # python3
+    from urllib.parse import urljoin, urlencode
+    from urllib.request import build_opener, Request
+    from urllib.error import URLError
+except ImportError:
+    # python2
+    from urlparse import urljoin
+    from urllib import urlencode
+    from urllib2 import build_opener, Request, URLError
+
+import numpy as np
+from sklearn.datasets.base import Bunch
+from sklearn.feature_extraction import DictVectorizer
+
+from .._utils.compat import _basestring
+from .utils import _fetch_file, _get_dataset_dir, _get_dataset_descr
+
+
+_NEUROVAULT_BASE_URL = 'http://neurovault.org/api/'
+_NEUROVAULT_COLLECTIONS_URL = urljoin(_NEUROVAULT_BASE_URL, 'collections/')
+_NEUROVAULT_IMAGES_URL = urljoin(_NEUROVAULT_BASE_URL, 'images/')
+_NEUROSYNTH_FETCH_WORDS_URL = 'http://neurosynth.org/api/v2/decode/'
+
+_COL_FILTERS_AVAILABLE_ON_SERVER = ('DOI', 'name', 'owner', 'id')
```

Comment:
the owner id (an int) is a filter that can be applied by the server itself. Of course the user can choose to filter based on the name; in this case it will be done locally. This is just the list of filters we can include in a URL, it is determined by Neurovault